central/scannerdefinitions/handler/handler.go

```diff
@@ -117,44 +118,33 @@ func (h *httpHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 func (h *httpHandler) get(w http.ResponseWriter, r *http.Request) {
+	// Open the most recent definitions file for the provided `uuid`.
 	uuid := r.URL.Query().Get(`uuid`)
-	if !h.online || uuid == "" {
-		// Default to the offline dump.
-		serveFile(w, r, h.offlineFile.Path())
-		return
-	}
-
-	u := h.getUpdater(uuid)
-	// Start may be called multiple times, but will only start the updater once.
-	u.Start()
-
-	// Serve the more recent of the requested file and the manually uploaded definitions.
-
-	onlineF, onlineModTime, err := u.file.Open()
+	f, modTime, err := h.openMostRecentDefinitions(uuid)
 	if err != nil {
-		writeErrorForFile(w, err, u.file.Path())
+		writeErrorForFile(w, err, uuid)
 		return
 	}
-	defer utils.IgnoreError(onlineF.Close)
 
-	offlineF, offlineModTime, err := h.offlineFile.Open()
-	if err != nil {
-		writeErrorForFile(w, err, h.offlineFile.Path())
+	// It is possible no offline Scanner definitions were uploaded, or Central cannot
+	// reach the definitions object, or there is no definitions for the given
+	// `uuid`; in any of those cases, `f` will be `nil`.
+	if f == nil {
+		writeErrorNotFound(w)
 		return
 	}
-	defer utils.IgnoreError(offlineF.Close)
 
-	f, modTime := onlineF, onlineModTime
-	if offlineModTime.After(onlineModTime) {
-		f, modTime = offlineF, offlineModTime
-	}
+	defer utils.IgnoreError(f.Close)
 
-	// It is possible no offline Scanner definitions are uploaded and Central cannot reach
-	// the vulnerability source or the client requested vulnerabilities which do not exist.
-	// Check for nil to protect against this.
-	if f == nil {
-		writeErrorNotFound(w)
-		return
+	// If `file` was provided, extract from definitions' bundle to a
+	// temporary file and serve that instead.
+	fileName := r.URL.Query().Get(`file`)
+	if fileName != "" {
+		f, err = openFromArchive(f.Name(), fileName)
```

Comment:
This endpoint does not add additional compression to the output, as it was traditionally solely used to return a ZIP file, so if we only return one file, then we may need to consider compressing the results